### PR TITLE
Enable Mono CI builds

### PR DESCRIPTION
**Pull request by radical**

### CODE REVIEW

File: netci.groovy [hunk 1]
```diff
@@ -11,6 +11,8 @@ def project = GithubProject
 
         if (osName == 'Windows_NT') {
             runtimes.add('Desktop')
+        } else {
+            runtimes.add('Mono')
         }
 
         // TODO: Mono
```
Comment (rainersigwald) [inline netci.groovy]: Delete stale comment 🎉

File: netci.groovy [hunk 2]
```diff
@@ -55,15 +57,23 @@ def project = GithubProject
                 case 'OSX':
                     newJob.with{
                         steps{
-                            shell("./cibuild.sh --scope Test --target ${runtime}")
+                            buildCmd="./cibuild.sh --scope Test --target ${runtime}"
+                            if (runtime == "Mono") {
+                                buildCmd += " --host Mono"
```
Comment (rainersigwald) [inline netci.groovy]: Isn't this implied by `--target`? I thought that was how we did it for `CoreCLR`.

Comment (radical) [inline netci.groovy]: `--target` selects the configuration (like `Debug-MONO`) and `--host` selects coreclr/mono to use for running msbuild.

File: netci.groovy [hunk 3]
```diff
@@ -11,6 +11,8 @@ def project = GithubProject
 
         if (osName == 'Windows_NT') {
             runtimes.add('Desktop')
+        } else {
+            runtimes.add('Mono')
```
Comment (radical) [inline netci.groovy]: I was going for the quickest route for now, as I thought that I will have to figure out mono install location etc. But we don't need to do that, we can build the mono configuration using .net itself. so, `./cibuild.sh --scope Test --target Mono` should be fine. I will add that tomorrrow and update the PR.

**Comment by radical** [comment]
Opening a new PR against master.

**Pull request closed without merging**